data/src/main/java/com/bottlerocketstudios/mapsdemo/data/utils/KotlinResultUtils.kt

```diff
@@ -0,0 +1,11 @@
+package com.bottlerocketstudios.mapsdemo.data.utils
+
+import retrofit2.HttpException
+
+fun <T> Result<T>.mapErrors(): Result<T> {
+    return when (val exception = exceptionOrNull()) {
+        null -> this
+        is HttpException -> Result.failure(exception.toApiException())
+        else -> Result.failure(exception)
+    }
+}
```

Comment:
@Josh-Gebbeken The else line value (`Result.failure(exception)`) is effectively the same as returning `this`. How about restructuring the logic to look like below:

```suggestion
fun <T> Result<T>.mapErrors(): Result<T> {
    return when (val exception = exceptionOrNull()) {
        is HttpException -> Result.failure(exception.toApiException())
        else -> this
    }
}
```